shared/modules/utility_module.go

```diff
@@ -26,17 +26,24 @@ type UtilityModule interface {
 type UtilityContext interface {
 	// Block operations
 
+	SetProposalBlock(blockHash string, proposerAddr []byte, transactions [][]byte) error
```

Comment:
Shouldn't we add a comment stating this should be only used by the proposer? Or is there a reason for validators to run this?

Reply:
Adding the following comment:

```
	// This function is intended to be called by any type of node during state transitions. For example,
	// both block proposers and verifiers will use it to create a context (before finalizing it) during consensus,
	// and all verifiers will call it during state sync.
```